src/CMakeLists.txt

```diff
@@ -0,0 +1,248 @@
+cmake_minimum_required(VERSION 3.12.0 FATAL_ERROR)
+
+include(GNUInstallDirs)
+
+add_library(
+    qofono-qt${QT_MAJOR_VERSION} SHARED 
+    qofono.cpp
+    qofonodbustypes.cpp
+    qofonoobject.cpp
+    qofonomanager.cpp
+    qofonomodem.cpp
+    qofonomodeminterface.cpp
+    qofonomodeminterface2.cpp
+    qofonoassistedsatellitenavigation.cpp
+    qofonoconnectionmanager.cpp
+    qofonoconnectioncontext.cpp
+    qofononetworkregistration.cpp
+    qofononetworkoperator.cpp
+    qofonocellbroadcast.cpp
+    qofonocallbarring.cpp
+    qofonocallforwarding.cpp
+    qofonocallmeter.cpp
+    qofonocallsettings.cpp
+    qofonocallvolume.cpp
+    qofonohandsfree.cpp
+    qofonohandsfreeaudioagent.cpp
+    qofonohandsfreeaudiocard.cpp
+    qofonohandsfreeaudiomanager.cpp
+    qofonoipmultimediasystem.cpp
+    qofonolocationreporting.cpp
+    qofonomessage.cpp
+    qofonomessagemanager.cpp
+    qofonomessagewaiting.cpp
+    qofonosimmanager.cpp
+    qofonosimwatcher.cpp
+    qofonophonebook.cpp
+    qofonopositioningrequestagent.cpp
+    qofonosmartmessaging.cpp
+    qofonosmartmessagingagent.cpp
+    qofonovoicecall.cpp
+    qofonovoicecallmanager.cpp
+    qofonoradiosettings.cpp
+    qofonosupplementaryservices.cpp
+    qofonotexttelephony.cpp
+)
+
+set(PUBLIC_HEADER_FILES
+    qofono.h
+    qofono_global.h
+    qofonoobject.h
+    qofonomanager.h
+    dbustypes.h
+    qofonomodem.h
+    qofonomodeminterface.h
+    qofonomodeminterface2.h
+    qofonoassistedsatellitenavigation.h
+    qofonoconnectionmanager.h
+    qofonoconnectioncontext.h
+    qofononetworkregistration.h
+    qofononetworkoperator.h
+    qofonocellbroadcast.h
+    qofonocallbarring.h
+    qofonocallforwarding.h
+    qofonoipmultimediasystem.h
+    qofonolocationreporting.h
+    qofonocallmeter.h
+    qofonocallsettings.h
+    qofonocallvolume.h
+    qofonohandsfree.h
+    qofonohandsfreeaudioagent.h
+    qofonohandsfreeaudiocard.h
+    qofonohandsfreeaudiomanager.h
+    qofonomessage.h
+    qofonomessagemanager.h
+    qofonomessagewaiting.h
+    qofonosimmanager.h
+    qofonosimwatcher.h
+    qofonophonebook.h
+    qofonopositioningrequestagent.h
+    qofonosmartmessaging.h
+    qofonosmartmessagingagent.h
+    qofonovoicecall.h
+    qofonovoicecallmanager.h
+    qofonoradiosettings.h
+    qofonosupplementaryservices.h
+    qofonotexttelephony.h
+)
+
+set_target_properties(
+    qofono-qt${QT_MAJOR_VERSION}
+    PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+
+target_compile_definitions(qofono-qt${QT_MAJOR_VERSION} PRIVATE QOFONO_LIBRARY)
+
+target_compile_options(qofono-qt${QT_MAJOR_VERSION} PRIVATE -O2 -Wno-unused-parameter -fvisibility=hidden -fvisibility-inlines-hidden)
```

Comment:
set_target_properties(<...> CXX_VISIBILITY_PRESET hidden) should be more standard way to do this?